pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -217,7 +173,30 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 				continue
 			}
 			clusterKey := buildClusterKey(service, port, cb, proxy, efKeys)
-			cached, allFound := cb.getAllCachedSubsetClusters(*clusterKey)
+			// clusterKey can't be nil as per buildClusterKey
+			subsetKeys := cb.getSubsetKeys(*clusterKey)
+			// it exists, we shouldn't mark as deleted
+			deleted.Delete(clusterKey.clusterName)
+			// we can only skip generation of the clusters associated with this service iff none of the updated
+			// resources depend on that cluster
+			canSkip := skipForDelta(clusterKey, updated) && skipForDelta(oldCacheKeys[clusterKey.clusterName], updated)
+			// the same applies for all subsets
+			for _, ss := range subsetKeys {
+				skipSubset := skipForDelta(&ss, updated) && skipForDelta(oldCacheKeys[ss.clusterName], updated)
+				deleted.Delete(ss.clusterName)
+				// We can only skip if we skip ALL subsets. This could probably be optimized in the future
+				canSkip = canSkip && skipSubset
+			}
+			// replace old cache key with our generated key
+			oldCacheKeys[clusterKey.clusterName] = clusterKey
+			for _, ss := range subsetKeys {
```

Comment:
I believe you can make the linter happy with the modification to the for loop as follows:
```
for i, ss := range subsetKeys {
     oldCacheKeys[ss.clusterName] = &subsetKeys[i]
 }
```